src/tools/compiletest/src/runtest.rs

```diff
@@ -2399,15 +2399,21 @@ actual:\n\
     fn normalize_output(&self, output: &str, custom_rules: &[(String, String)]) -> String {
         let parent_dir = self.testpaths.file.parent().unwrap();
         let cflags = self.props.compile_flags.join(" ");
-        let parent_dir_str = if cflags.contains("--error-format json")
-                             || cflags.contains("--error-format pretty-json") {
+        let json = cflags.contains("--error-format json") ||
+                   cflags.contains("--error-format pretty-json");
+        let parent_dir_str = if json {
             parent_dir.display().to_string().replace("\\", "\\\\")
         } else {
             parent_dir.display().to_string()
         };
 
-        let mut normalized = output.replace(&parent_dir_str, "$DIR")
-              .replace("\\\\", "\\") // denormalize for paths on windows
+        let mut normalized = output.replace(&parent_dir_str, "$DIR");
+
+        if json {
+            normalized = normalized.replace("\\n", "\n"); // verbatim newline in json strings
```

Comment:
Could you explain this in short? This text is going to be used as a whole without any preprocessing?

Reply:
We just want this so the json looks readable if suggestions contain multilines. As a side effect error explanations also become readable

Reply:
I added a comment explaining what's going on